benchmark/mnist_benchmark_visualiser.py

```diff
@@ -0,0 +1,167 @@
+# © Crown Copyright GCHQ
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Visualise the results of ``mnist_benchmark.py``."""
+
+import json
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def load_benchmark_data(filename: str) -> dict:
+    """
+    Load benchmark data from a specified JSON file.
+
+    :param filename: The path to the JSON file containing benchmark data.
+    :return: A dictionary containing the benchmark data, or an empty dictionary
+             if there was an error loading the file.
+    """
+    try:
+        with open(filename, "r", encoding="utf-8") as file:
+            return json.load(file)
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        print(f"Error loading the file: {e}")
+        return {}
+
+
+def compute_statistics(data_by_solver: dict, coreset_sizes: list[int]) -> tuple[dict]:
```

Comment:
Fix return type - I think it should be  `tuple[dict[Any, dict[str, list]], dict[Any, dict[str, list]]]` but not certain